sockeye/data_io.py

```diff
@@ -475,6 +575,21 @@ def get_num_shards(num_samples: int, samples_per_shard: int, min_num_shards: int
     :param min_num_shards: Minimum number of shards.
     :return: Number of shards.
     """
+    if curriculum_score_file and check_score_file(num_samples, curriculum_score_file):
+        with open(curriculum_score_file, 'r') as curriculum_score_file:
+            sample_scores = [int(line.strip()) for line in curriculum_score_file.readlines() if line.strip()]
```

Comment:
If you really need to read in the file this many times for the feature, it might make sense to write a little generator that yields the integers from the file:
```python
def sample_score_iterator(fname):
  with smart_open(fname, 'r') as fin:
    for line in fin:
      yield int(line.strip())
```